fe/fe-core/src/main/java/com/starrocks/connector/unified/UnifiedConnector.java

```diff
@@ -0,0 +1,91 @@
+import com.starrocks.connector.ConnectorMetadata;
+import com.starrocks.connector.config.ConnectorConfig;
+import com.starrocks.connector.delta.DeltaLakeConnector;
+import com.starrocks.connector.hive.HiveConnector;
+import com.starrocks.connector.hudi.HudiConnector;
+import com.starrocks.connector.iceberg.IcebergConnector;
+import com.starrocks.credential.CloudConfiguration;
+import com.starrocks.sql.analyzer.SemanticException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.starrocks.catalog.Table.TableType.DELTALAKE;
+import static com.starrocks.catalog.Table.TableType.HIVE;
+import static com.starrocks.catalog.Table.TableType.HUDI;
+import static com.starrocks.catalog.Table.TableType.ICEBERG;
+import static com.starrocks.connector.hive.HiveConnector.HIVE_METASTORE_TYPE;
+import static com.starrocks.connector.iceberg.IcebergConnector.ICEBERG_CATALOG_TYPE;
+
+public class UnifiedConnector implements Connector {
+    public static final String UNIFIED_METASTORE_TYPE = "unified.metastore.type";
+    public static final List<String> SUPPORTED_METASTORE_TYPE = ImmutableList.of("hive", "glue");
+    private final Map<Table.TableType, Connector> connectorMap;
+
+    public UnifiedConnector(ConnectorContext context) {
+        String metastoreType = context.getProperties().get(UNIFIED_METASTORE_TYPE);
+        if (!SUPPORTED_METASTORE_TYPE.contains(metastoreType)) {
+            throw new SemanticException("Unified catalog only supports hive and glue as metastore.");
+        }
+
+        ImmutableMap.Builder<String, String> derivedProperties = ImmutableMap.builder();
+        derivedProperties.putAll(context.getProperties());
+        derivedProperties.put(HIVE_METASTORE_TYPE, metastoreType);
+        derivedProperties.put(ICEBERG_CATALOG_TYPE, metastoreType);
+
+        ConnectorContext derivedContext = new ConnectorContext(context.getCatalogName(), context.getType(),
+                derivedProperties.build());
+
+        connectorMap = ImmutableMap.of(
+                HIVE, new HiveConnector(derivedContext),
+                ICEBERG, new IcebergConnector(derivedContext),
+                HUDI, new HudiConnector(derivedContext),
+                DELTALAKE, new DeltaLakeConnector(derivedContext)
+        );
+    }
+
+    @Override
+    public ConnectorMetadata getMetadata() {
+        Map<Table.TableType, ConnectorMetadata> metadataMap = new HashMap<>();
+        connectorMap.forEach((k, v) -> metadataMap.put(k, v.getMetadata()));
+
+        return new UnifiedMetadata(metadataMap);
+    }
+
+    @Override
+    public void shutdown() {
+        connectorMap.forEach((k, v) -> v.shutdown());
+    }
+
+    @Override
+    public void bindConfig(ConnectorConfig config) {
+        connectorMap.forEach((k, v) -> v.bindConfig(config));
+    }
+
+    @Override
+    public CloudConfiguration getCloudConfiguration() {
+        return connectorMap.get(HIVE).getCloudConfiguration();
```

Comment:
why only use HIVE?

Reply:
Cloud configurations are the same for any sub-connector